tests/rados/test_election_strategies.py

```diff
@@ -222,3 +283,382 @@ def check_mon_status(rados_obj, mon_obj):
             return False
         log.info(f"Mon {mon} is in healthy state: {status} | {desc}")
     return True
+
+
+def add_all_mons_to_disallow_list(mon_obj) -> None:
+    """Method to validate last Mon cannot be added to disallowed leaders list
+
+    Steps:
+    - Get list of all Mons from the cluster
+    - Add all Mons to disallowed leaders list
+    - Validate last allowed Mon cannot be added to disallow list
+    - Revert cluster back to original state by removing all Mons from disallow list
+
+    Args:
+        mon_obj: object MonElectionStrategies
+    Returns:
+        Raises Exception when test fails
+    Example:
+
+    """
+    log.info(
+        "\n -------------------------------------------"
+        "\n Starting validation:"
+        "\n `last allowed Mon` cannot be added to disallow list"
+        "\n -------------------------------------------"
+    )
+
+    mons = list(mon_obj.get_mon_quorum().keys())
+    last_allowed_mon = mons.pop()
+
+    for mon in mons:
+        log.info(f"Adding mon {mon} to disallow list")
+        if not mon_obj.set_disallow_mon(mon=mon):
+            raise AssertionError(f"could not add Mon {mon} to disallowed list")
```

Comment:
Check if the mon daemon listed via dump is down, when we hit a assertion here. fail when mon is down and we add it to disallowed list.